kms/src/main/java/io/kroxylicious/kms/provider/kroxylicious/inmemory/InMemoryEdek.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.kms.provider.kroxylicious.inmemory;
+
+import java.util.Arrays;
+import java.util.Objects;
+
+record InMemoryEdek(
+                    int numAuthBits,
+                    byte[] iv,
+                    byte[] edek) {
```

Comment:
add canonical ctor + validate the arguments

Reply:
I don't mind validating the auth bits because -138D defines what we should use. But it's a bit more ambiguous about IV length in general. Yes it's best if it's 96 bits, but should we enforce that here. It's only code for testing...?